### ISSUE TITLE
sshfs hangs macos on "resource busy" — forces reboot

### ISSUE THREAD

**Issue by 0x6a77** [telepresenceio/telepresence#2033]
if you find that `telepresence intercept` on a mac freezes file system access, forcing a reboot, where the root cause is `sshfs` freezes on resource busy, the ultimate cause may be that your target pod's `traffic-agent` has mounted one of your volumes in `tel_app_mounts` with `df: /tel_app_mounts/<VOLUME>: Permission denied`

the traffic agent should not try to mount volumes where it doesn't have permission equal to the desired end state permission, e.g. read, read-write. it should instead log the reason it can't mount them.

a workaround for this problem is to figure out why the traffic agent can't read the target workload's volume. likely it's a difference in permissions. e.g. you've set the pod to run non-root, so the traffic agent does this, but then it can't read the volume. there could be many reason why traffic-agent can't read the volume, but if you sort this out, sshfs (on mac) won't freeze the local machine. (this bug report is to at least prevent the freeze situation so users can then explore why their volume didn't mount without the hindrance of frozen machine.)

**Comment by 0x6a77** [comment]
btw, since someone will have to touch this part of the code (and test that sshfs doesn't hang), we can also use this as a chance to reduce overall sshfs hang-failures for other causes like network latency/reliability. adding `-o reconnect,ServerAliveInterval=1` to the sshfs invocation probably will increase stability for a tiny performance cost

**Comment by xavidop** [comment]
Hello!

we are facing this issue as well!

this is the log from a `traffic-agent`:

```
time="2021-09-27 11:24:20.2133" level=trace msg="Logging at this level (LOG_LEVEL=\"trace\")" func=main.makeBaseLogger file="github.com/telepresenceio/telepresence/v2/cmd/traffic/logger.go:39"
time="2021-09-27 11:24:20.2135" level=info msg="Traffic Agent v2.4.1 [pid:1]" func=github.com/telepresenceio/telepresence/v2/cmd/traffic/cmd/agent.Main file="github.com/telepresenceio/telepresence
/v2/cmd/traffic/cmd/agent/agent.go:145"
time="2021-09-27 11:24:20.2136" level=info msg="{Name:ingest Namespace:dev-xavi PodIP:10.127.137.58 AgentPort:9900 AppMounts:/tel_app_mounts AppPort:8014 ManagerHost:traffic-manager.ambassador ManagerPort:8081}" func=github.com/telepresenceio/telepresence/v2/cmd/traffic/cmd/agent.Main file="github.com/telepresenceio/telepresence/v2/cmd/traffic/cmd/agent/agent.go:164"
time="2021-09-27 11:24:20.2140" level=debug msg="checking agent secrets mount path: /var/run/secrets/kubernetes.io" func="github.com/telepresenceio/telepresence/v2/cmd/traffic/cmd/agent.(*Config).AddSecretsMounts" file="github.com/telepresenceio/telepresence/v2/cmd/traffic/cmd/agent/agent.go:107"
time="2021-09-27 11:24:20.2140" level=debug msg="checking appmounts directory: /var/run/secrets/kubernetes.io" func="github.com/telepresenceio/telepresence/v2/cmd/traffic/cmd/agent.(*Config).AddSecretsMounts" file="github.com/telepresenceio/telepresence/v2/cmd/traffic/cmd/agent/agent.go:114"
time="2021-09-27 11:24:20.2141" level=debug msg="create appmounts directory: /tel_app_mounts/var/run/secrets/kubernetes.io" func="github.com/telepresenceio/telepresence/v2/cmd/traffic/cmd/agent.(*Config).AddSecretsMounts" file="github.com/telepresenceio/telepresence/v2/cmd/traffic/cmd/agent/agent.go:120"
time="2021-09-27 11:24:20.2142" level=error msg="There was a problem with agent mounts: mkdir /tel_app_mounts/var: permission denied" func=github.com/telepresenceio/telepresence/v2/cmd/traffic/cmd/agent.Main file="github.com/telepresenceio/telepresence/v2/cmd/traffic/cmd/agent/agent.go:190"
time="2021-09-27 11:24:20.2142" level=debug msg="agent mount paths: /app/config/:/var/run/secrets/kubernetes.io/serviceaccount" func="github.com/telepresenceio/telepresence/v2/cmd/traffic/cmd/agent.(*Config).HasMounts" file="github.com/telepresenceio/telepresence/v2/cmd/traffic/cmd/agent/agent.go:81"
time="2021-09-27 11:24:20.2152" level=debug msg="Forwarding from :9900" func="github.com/telepresenceio/telepresence/v2/pkg/forwarder.(*Forwarder).ServeListener" file="github.com/telepresenceio/telepresence/v2/pkg/forwarder/forwarder.go:63" THREAD=/forward
time="2021-09-27 11:24:25.2224" level=info msg="Connected to Manager v2.4.4-39-g4049395a-1632433143" func=github.com/telepresenceio/telepresence/v2/cmd/traffic/cmd/agent.TalkToManager file="github.com/telepresenceio/telepresence/v2/cmd/traffic/cmd/agent/client.go:53" THREAD=/client
time="2021-09-27 11:24:
```
Here is the full log [out.log](https://github.com/telepresenceio/telepresence/files/7237960/out.log)


So having that issue, our laptops stop working and the only way to have everything up and running is just restarting the laptop.

I had to rollback to 2.4.1 version on the traffic-manager and it seems working again

**Comment by vl-shopback** [comment]
is there any update

**Comment by cindymullins-dw** [comment]
Resolved in v2.5.  Thanks, @xavidop !
